homeassistant/components/zha/config_flow.py

```diff
@@ -0,0 +1,53 @@
+"""Config flow for ZHA."""
+from collections import OrderedDict
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from .helpers import check_zigpy_connection
+from .const import (
+    DOMAIN, CONF_RADIO_TYPE, CONF_USB_PATH, RadioType
+)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ZhaFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    async def async_step_user(self, user_input=None):
+        """Handle a zha config flow start."""
+        if self._async_current_entries():
+            return self.async_abort(reason='single_instance_allowed')
+
+        errors = {}
+
+        fields = OrderedDict()
+        fields[vol.Required(CONF_USB_PATH)] = str
+        fields[vol.Optional(CONF_RADIO_TYPE, default='ezsp')] = vol.In(
+            RadioType.list()
+        )
+
+        if user_input is not None:
+            test = await check_zigpy_connection(user_input[CONF_USB_PATH],
```

Comment:
We may want to wrap this awaitable into `asyncio.wait_for()` with a timeout, as in my tests if you give it a valid serial port with nothing connected to it, then "ezsp" radio would wait indefinitely waiting for an "init" frame response. eg
```
2018-11-25 11:07:54 DEBUG (MainThread) [zigpy.appdb] Attribute id: 0 value: 204
2018-11-25 11:07:54 DEBUG (MainThread) [bellows.uart] Sending: b'1ac038bc7e'
```
for XBee radio we'd need about a 30s timeout, as XBee would scan all baudrates trying to properly reconfigure XBee API mode.